bot/storage/storage.py

```diff
@@ -0,0 +1,61 @@
+import datetime
+import logging
+
+class Storage(object):
+    """ Virtual Base Class for Storage Implementations.
```

Comment:
This jargon isn't Python-esque, should probably be changed to the equivalent (whatever that is, I don't know exactly)

Reply:
abstract parent class?  requires child completion class ? :)  I'll give it some thought.

Reply:
"requires child completion class" is my favourite.